### DISCUSSION TITLE
[Forge 1.20.1 Suggestion] Currency option for rewards, and being able to make unitWorth be the per-item value

### DISCUSSION THREAD

**Discussion by vizthex123** [ejektaflex/Bountiful#274] [Mod Ideas]
The 1.12 version of the mod had a config option that made rewards act more like a currency rather than picking more or less at random, and it'd be nice to have it back.

For example, you could set Emeralds to have a value of 500 and Diamonds to have a value of 1,000, and rewards would be selected starting from the highest value to lowest value rather than being random.

In addition, being able to define `unitWorth` as the value of each item (for objectives, rewards, or both) rather than the entire thing would let modpack devs better utilize a currency-focused reward system (e.g. a reward range of 2 - 4 can be worth 400 in the current system, but with the currency system enabled each item would be worth 400 - making the total 800 - 1600).

A config option to disable the default recipes would be nice too. I did manually remove them, but figured I'd append it this post.

**Comment by ejektaflex** [comment]
I'm a bit confused about your second point. What you're suggesting is already in place - `unitWorth` is the value of one *unit* of the item (e.g. for a reward range of 5-10 and a unitWorth of 30 the entire reward range for that item would be 150-300). This is covered in the [wiki](https://kambrik.ejekta.io/mods/bountiful/advanced/generation). 

As for the first point, this would require a bit of work since the matching algorithm is so much more complicated than it was in 1.12. I'll have to see if it's feasibly something we can switch to without too much work!

**Reply by vizthex123** [reply]
ah, i must've misread it then lol.

but yeah, if you could look into the currency reward option it'd be nice. Even with rarities and weight multipliers, it still picks rewards basically randomly. Just a minor nuisance at best though, but I figured i'd suggest it.

would also be nice if you could make rarities work when lowercase so it matches the other parameters.
